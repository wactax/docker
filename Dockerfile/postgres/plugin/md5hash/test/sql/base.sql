CREATE EXTENSION md5hash;

SELECT md5('A')::md5hash;
SELECT md5('B')::md5hash;
SELECT md5('A')::md5hash = md5('A')::md5hash;
SELECT md5('A')::md5hash = md5('B')::md5hash;

SELECT md5('A')::md5hash <= md5('B')::md5hash;
SELECT md5('A')::md5hash <  md5('B')::md5hash;
SELECT md5('A')::md5hash >= md5('B')::md5hash;
SELECT md5('A')::md5hash >  md5('B')::md5hash;

CREATE TABLE test ( id md5hash );
INSERT INTO test SELECT md5(i::text) FROM generate_series(1,10000) s(i);
ANALYZE test;
SELECT relpages FROM pg_class WHERE relname = 'test';