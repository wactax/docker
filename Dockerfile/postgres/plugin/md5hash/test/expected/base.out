CREATE EXTENSION md5hash;
SELECT md5('A')::md5hash;
               md5                
----------------------------------
 7fc56270e7a70fa81a5935b72eacbe29
(1 row)

SELECT md5('B')::md5hash;
               md5                
----------------------------------
 9d5ed678fe57bcca610140957afab571
(1 row)

SELECT md5('A')::md5hash = md5('A')::md5hash;
 ?column? 
----------
 t
(1 row)

SELECT md5('A')::md5hash = md5('B')::md5hash;
 ?column? 
----------
 f
(1 row)

SELECT md5('A')::md5hash <= md5('B')::md5hash;
 ?column? 
----------
 t
(1 row)

SELECT md5('A')::md5hash <  md5('B')::md5hash;
 ?column? 
----------
 t
(1 row)

SELECT md5('A')::md5hash >= md5('B')::md5hash;
 ?column? 
----------
 f
(1 row)

SELECT md5('A')::md5hash >  md5('B')::md5hash;
 ?column? 
----------
 f
(1 row)

CREATE TABLE test ( id md5hash );
INSERT INTO test SELECT md5(i::text) FROM generate_series(1,10000) s(i);
ANALYZE test;
SELECT relpages FROM pg_class WHERE relname = 'test';
 relpages 
----------
       55
(1 row)

